### ISSUE TITLE
How can i customize the toolbar?

### ISSUE THREAD

**Issue by FrancescoMussi** [winkerVSbecks/angular-pdf-viewer#15]
Hello!
Good job!
I just don't understand from the docs how can i customize the template.
For example i would like to remove:  Zoom in - Zoom out - Rotate
and modify Back and Next putting two icons instead of the text

How can i do that?

Thank you very much!

**Comment by winkerVSbecks** [comment]
What I meant was that you do not have to use the default toolbar at all. In most cases you probably want to build something custom and probably use your own CSS. 

You can do so by first disabling the default toolbar, like so:

```html
<pdf-viewer
    delegate-handle="my-pdf-container"
    show-toolbar="false"></pdf-viewer>
```

Your own toolbar can then use the [Delegate Service](https://github.com/winkerVSbecks/angular-pdf-viewer#delegate-service-pdfdelegate) to control the PDF.

The actual toolbar is like 40 lines of code. You can use it for reference: https://github.com/winkerVSbecks/angular-pdf-viewer/blob/master/src/js/pdf-viewer-toolbar.js

**Comment by FrancescoMussi** [comment]
Thank for reply.
I have tried but it doesn't work. I am doing something wrong.
I have add in my app.js these lines:

angular.module('pdf', [])
  .service('pdfDelegate', delegateService([
    'prev',
    'next',
    'getPageCount',
    'getCurrentPage',
    'goToPage',
    'load'
  ]));


angular.module('pdf')
  .directive('pdfViewerToolbar', [
    'pdfDelegate',
  function(pdfDelegate) {
    return {
      restrict: 'E',
      template:
        '<div class="clearfix mb2 white bg-blue">' +
          '<div class="left">' +
            '<a href=""' +
              'ng-click="prev()"' +
              'class="button py2 m0 button-nav-dark">Back' +
            '</a>' +
            '<a href=""' +
              'ng-click="next()"' +
              'class="button py2 m0 button-nav-dark">Next' +
            '</a>' +
            '<span class="px1">Page</span> ' +
            '<input type="text" class="field-dark" ' +
              'min=1 ng-model="currentPage" ng-change="goToPage()" ' +
               'style="width: 10%"> ' +
            ' / {{pageCount}}' +
          '</div>' +
        '</div>',
      scope: { pageCount: '=' },
      link: function(scope, element, attrs) {
        var id = attrs.delegateHandle;
        scope.currentPage = 1;

        scope.prev = function() {
          pdfDelegate
            .$getByHandle(id)
            .prev();
          updateCurrentPage();
        };
        scope.next = function() {
          pdfDelegate
            .$getByHandle(id)
            .next();
          updateCurrentPage();
        };
        scope.goToPage = function() {
          pdfDelegate
            .$getByHandle(id)
            .goToPage(scope.currentPage);
        };

        var updateCurrentPage = function() {
          scope.currentPage = pdfDelegate
                                .$getByHandle(id)
                                .getCurrentPage();
        };
      }
    };
}]);


But i have obtained something near to what i need by removing the link to Zoom in - Zoom out - Rotate from angular-pdf-viewer.min.js

**Comment by winkerVSbecks** [comment]
Two things:
- You don't need to replicate the delegate service. 
- Use your own module and inject the pdf module into it

```js
angular.module('myApp', [ 'pdf' ])
  .directive('myPdfViewerToolbar', [
     'pdfDelegate',
  function(pdfDelegate) {
    return {
       restrict: 'E',
       template:
        '<div class="clearfix mb2 white bg-blue">' +
          '<div class="left">' +
            '<a href=""' +
              'ng-click="prev()"' +
              'class="button py2 m0 button-nav-dark">Back' +
             '</a>' +
            '<a href=""' +
              'ng-click="next()"' +
              'class="button py2 m0 button-nav-dark">Next' +
            '</a>' +
            '<span class="px1">Page</span> ' +
            '<input type="text" class="field-dark" ' +
              'min=1 ng-model="currentPage" ng-change="goToPage()" ' +
               'style="width: 10%"> ' +
            ' / {{pageCount}}' +
          '</div>' +
        '</div>',
       scope: { pageCount: '=' },
       link: function(scope, element, attrs) {
         var id = attrs.delegateHandle;
         scope.currentPage = 1;

        scope.prev = function() {
          pdfDelegate
            .$getByHandle(id)
            .prev();
          updateCurrentPage();
        };
        scope.next = function() {
          pdfDelegate
            .$getByHandle(id)
            .next();
          updateCurrentPage();
        };
        scope.goToPage = function() {
          pdfDelegate
             .$getByHandle(id)
             .goToPage(scope.currentPage);
         };

        var updateCurrentPage = function() {
          scope.currentPage = pdfDelegate
                                .$getByHandle(id)
                                 .getCurrentPage();
        };
      }
    };
}]);
```


and use it like so:

```html
<my-pdf-viewer-toolbar
    delegate-handle="my-pdf-container"></my-pdf-viewer-toolbar>
<pdf-viewer
    delegate-handle="my-pdf-container"
    url="pdfUrl"
    scale="1"
    show-toolbar="false"></pdf-viewer>
```

**Comment by FrancescoMussi** [comment]
Ok, thank you very much!
Now the toolbar is working perfectly!

**Comment by newrdev** [comment]
Hi,
Thank you for making such an awesome tool and sorry for jumping into an already closed issue.

I am attempting to make my own toolbar following the instructions here. The only issue i am having, is that i cannot get the pageCount on the custom toolbar after i load a new pdf url.
I am new at AngularJS so please bear with me. But i had a few questions. Is it possible for maybe the load function to return a promise? At the moment i do not see any way of knowing when the pdf has finished loading so i can get the pageCount or maybe let the user know it is loading.

Thanks in advance.

**Comment by winkerVSbecks** [comment]
You can get the page count like so:

```js
pdfDelegate
  .$getByHandle(id)
  .getPageCount()
```

**Comment by newrdev** [comment]
@winkerVSbecks thanks for the reply. 

That is how i am currently retrieving it, but, how can i get the pageCount after reloading the pdfUrl? Currently i need to wrap it in a $timeout and guess the time it will take to load, then after that i can get the pageCount.

**Comment by winkerVSbecks** [comment]
Yea that's a good point. There doesn't seem to be a clean way of getting the `pageCount` if you reload the `pdfUrl`. 

Could you test something for me? In the directive controller: https://github.com/winkerVSbecks/angular-pdf-viewer/blob/master/src/js/pdf-ctrl.js… If you return the `PDFJS` promise in `self.load` you should be able to use it as following:

```js
var myPdf  = pdfDelegate.$getByHandle('my-pdf-container');

myPdf
  .load('url-of-the-new-file.pdf')
  .then(function()) {
        myPdf. getPageCount()
   });
```

**Comment by newrdev** [comment]
@winkerVSbecks easy fix. That works perfectly!

**Comment by winkerVSbecks** [comment]
@newrdev awesome! You can submit a PR if you have time. Otherwise I'll update and release a new version tomorrow. Thanks

**Comment by newrdev** [comment]
@winkerVSbecks PR submitted.

**Comment by winkerVSbecks** [comment]
@newrdev thanks. I made a new release. You can run `bower update` to get the latest version.

**Comment by newrdev** [comment]
@winkerVSbecks thanks you!
I don't know if I'm overstepping here, but this fix could also be a solution for #13 .
Just a thought. Thanks once again!

**Comment by winkerVSbecks** [comment]
That's a good point. Thanks!
